staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go

```diff
@@ -187,13 +199,29 @@ func (az *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID stri
 
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (az *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
+	if providerID == "" {
+		return false, nil
+	}
+
 	nodeName, err := az.vmSet.GetNodeNameByProviderID(providerID)
 	if err != nil {
+		// returns false, because otherwise node is not deleted from cluster
```

Comment:
use because or otherwise?

Reply:
good catch. actually, the first sentence should be removed.